block/src/Pos/Block/Worker.hs

```diff
@@ -105,6 +107,8 @@ informerWorker =
 
 blkCreatorWorker :: BlockWorkMode ctx m => (WorkerSpec m, OutSpecs)
 blkCreatorWorker =
+    -- TODO: [CSL-198] Catch and report non-critical errors, as
+    -- 'onNewSlot' no longer swallows all exception and rethrows them instead.
```

Comment:
This worker calls `Diffusion.announceBlockHeader`, so I guess it may fail when something goes wrong with network. I propose to catch exceptions from `announceBlockHeader` and retry later if they occur. On other cases let's stop the program.